File: FTAnalyser/Core/ParentalGroup.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Xml;

namespace FTAnalyzer
{
    public class ParentalGroup {

        public Individual Individual { get; private set; }
        public Individual Father { get; private set; }
        public Individual Mother { get; private set; }
        public Fact ParentsMarriage { get; private set; }
        
        public ParentalGroup (Individual i, Individual f, Individual m, Fact marriage) {
            this.Individual = i;
            this.Father = f;
            this.Mother = m;
            this.ParentsMarriage = marriage;
        }

        #region Properties 

        public string Residence {
            get
            {
                Fact residence = Individual.GetPreferredFact(Fact.RESIDENCE);
                return (residence == null) ? "" : residence.Place;
            }
        }
        
        public string FathersName {
            get { return (Father == null) ? ""  : Father.Name; }
        }

        public string MothersName {
            get { return (Mother == null) ? ""  : Mother.Name; }
        }

        public string FathersOccupation {
            get { return (Father == null) ? ""  : Father.Occupation; }
        }

        public string MothersOccupation {
            get { return (Mother == null) ? ""  : Mother.Occupation; }
        }
        
        public string ParentsMarriageDate {
            get { return (ParentsMarriage == null) ? "" : ParentsMarriage.FactDate.DateString; }
        }
        
        public string ParentsMarriageLocation {
            get { return (ParentsMarriage == null) ? "" : ParentsMarriage.Place; }
        }

        #endregion

        public FactLocation BestLocation(FactDate when)
        {
            FactLocation i = Individual.BestLocation(when);
            if (ParentsMarriage == null)
                return i;
            FactLocation f = FactLocation.GetLocation(ParentsMarriage.Place);
            if (f.Level > i.Level)
                return f;
            else
                return i;
        }

        public string IsFatherDeceased(FactDate when)
        {
            return (Father == null || !Father.IsDeceased(when)) ? "" : "(Deceased)";
        }

        public string IsMotherDeceased(FactDate when)
        {
            return (Mother == null || !Mother.IsDeceased(when)) ? "" : "(Deceased)";
        }

        public Fact GetPreferredFact(string factType)
        {
            return Individual.GetPreferredFact(factType);
        }

        public FactDate GetPreferredFactDate(string factType)
        {
            return Individual.GetPreferredFactDate(factType);
        }

    }
}

File: FTAnalyser/Old Java Code/org/hardgate/familytree/core/ParentalGroup.java
/*
 * Created on 02-Jan-2005
 *
 */
package org.hardgate.familytree.core;

import java.io.Serializable;

import org.apache.log4j.Logger;

/**
 * @author db
 *
 */
public class ParentalGroup implements Serializable {

	private static Logger logger = Logger.getLogger(ParentalGroup.class);
	private static final long serialVersionUID = 0;
	
    private Individual individual;
    private Individual father, mother;
    private Fact parentsMarriage;
    
    public ParentalGroup (Individual i, Individual f, Individual m, Fact marriage) {
        this.individual = i;
        this.father = f;
        this.mother = m;
        this.parentsMarriage = marriage;
    }

    /**
     * @return Returns the father.
     */
    public Individual getFather () {
        return father;
    }
    /**
     * @return Returns the individual.
     */
    public Individual getIndividual () {
        return individual;
    }
    /**
     * @return Returns the mother.
     */
    public Individual getMother () {
        return mother;
    }
    /**
     * @return Returns the parentsMarriage.
     */
    public Fact getParentsMarriage () {
        return parentsMarriage;
    }

    public String getResidence() {
        Fact residence = individual.getPreferredFact(Fact.RESIDENCE);
        return (residence == null) ? "" : residence.getLocation();
    }
    
    public Fact getPreferredFact(String factType) {
        return individual.getPreferredFact(factType);
    }
    
    public FactDate getPreferredFactDate (String factType) {
        return individual.getPreferredFactDate(factType);
    }
    
    public String getFathersName () {
        return (father == null) ? ""  : father.getName();
    }

    public String getMothersName () {
        return (mother == null) ? ""  : mother.getName();
    }

    public String getFathersOccupation () {
        return (father == null) ? ""  : father.getOccupation();
    }

    public String getMothersOccupation () {
        return (mother == null) ? ""  : mother.getOccupation();
    }
    
    public String getFatherDeceased (FactDate when) {
        return (father == null || ! father.isDeceased(when)) ? "" : "(Deceased)"; 
    }

    public String getMotherDeceased (FactDate when) {
        return (mother == null || ! mother.isDeceased(when)) ? "" : "(Deceased)"; 
    }
    
    public String getParentsMarriageDate () {
        return (parentsMarriage == null) ? "" :
            	parentsMarriage.getFactDate().getDateString();
    }
    
    public String getParentsMarriageLocation () {
        return (parentsMarriage == null) ? "" :
        	parentsMarriage.getLocation();
    }
    
    public Location getBestLocation() {
        Location i = individual.getBestLocation();
        if (parentsMarriage == null)
            return i;
        Location f = new Location(parentsMarriage.getLocation());
        if (f.getLevel() > i.getLevel())
            return f;
        else
            return i;
    }
}
